### ISSUE TITLE
no-danger-with-children false positive with newline in element

### ISSUE THREAD

**Issue by mockdeep** [yannickcr/eslint-plugin-react#1262]
The following reports an error on the `no-danger-with-children` rule:

```jsx
<div dangerouslySetInnerHTML={'<span>stuff</span>'}>
</div>
```

I would expect if there is a newline between the opening and closing tag that the rule would not complain.

**Comment by ljharb** [comment]
That's still considered "children", afaik - why not `<div dangerouslySetInnerHTML={'<span>stuff</span>'} />`?

**Comment by mockdeep** [comment]
Ah, I didn't realize we could do self-closing divs in jsx. Still, I don't think newlines between tags are considered children. Hacked together this quick test:

```jsx
import React from 'react';
import { mount } from 'enzyme';

class Parent extends React.Component {

  render() {
    console.log(this.props.children);
    return <div>{this.props.children}</div>;
  }

}

it('does stuff', () => {
  mount(<Parent></Parent>); // => `undefined`
  mount(<Parent>
      </Parent>); // => `undefined`
  mount(<Parent />); // => `undefined`
  mount(<Parent> </Parent>); // => `' '`
  mount(<Parent>

      </Parent>); // => `undefined`
  mount(<Parent>foo</Parent>); // => `'foo'`
})
```

**Comment by ljharb** [comment]
Fair point - both your example and https://jsfiddle.net/ljharb/69z2wepo/80986/ implies it's a bug in eslint-plugin-react.

**Comment by mockdeep** [comment]
Ha, nice. I started typing `ReactDOM` and then fell back to what I knew off the top of my head.
